server/src/main/java/org/opensearch/index/translog/transfer/BlobStoreTransferService.java

```diff
@@ -48,7 +48,7 @@ public void uploadBlobAsync(
     ) {
         assert remoteTransferPath instanceof BlobPath;
         BlobPath blobPath = (BlobPath) remoteTransferPath;
-        executorService.execute(ActionRunnable.wrap(listener, l -> {
+        threadPool.executor(ThreadPool.Names.TRANSLOG_TRANSFER).execute(ActionRunnable.wrap(listener, l -> {
```

Comment:
What do we gain by replacing `executorService` by `threadpool`? We are anyway accessing only `TRANSLOG_TRANSFER` threadpool. 

When I saw separate threadpools for transfer and delete and we are passing `Threadpool` to BlobStoreTransferService, I assumed that we will access both the threadpools in this class in the respective methods.

Reply:
I would prefer the control of which thread pools to use to be controlled centrally in `TranslogTransferManager`

Reply:
Have made the interface consistent. Incorporated.